XDMA/linux-kernel/xdma/xdma_netdev.c

```diff
@@ -0,0 +1,307 @@
+#include "xdma_netdev.h"
+#include "xdma_mod.h"
+#include "cdev_sgdma.h"
+#include "libxdma.h"
+
+/*
+static void packet_dump(unsigned char *buf, int len)
+{
+        int i;
+        for (i = 0; i < len; i++) {
+                if (i % 16 == 0) pr_err("\n");
+                pr_err("%02x ", buf[i]); 
+        }
+        pr_err("\n");
+}
+*/
+
+static int check_transfer_align_packet(struct xdma_engine *engine,
+	const char __kernel *buf, size_t count, loff_t pos, int sync)
+{
+	if (!engine) {
+		pr_err("Invalid DMA engine\n");
+		return -EINVAL; }
+	/* AXI ST or AXI MM non-incremental addressing mode? */
+	if (engine->non_incr_addr) {
+		int buf_lsb = (int)((uintptr_t)buf) & (engine->addr_align - 1);
+		size_t len_lsb = count & ((size_t)engine->len_granularity - 1);
+		int pos_lsb = (int)pos & (engine->addr_align - 1);
+
+		dbg_tfr("AXI ST or MM non-incremental\n");
+		dbg_tfr("buf_lsb = %d, pos_lsb = %d, len_lsb = %ld\n", buf_lsb,
+			pos_lsb, len_lsb);
+
+		if (buf_lsb != 0) {
+			dbg_tfr("FAIL: non-aligned buffer address %p\n", buf);
+			return -EINVAL;
+		}
+
+		if ((pos_lsb != 0) && (sync)) { dbg_tfr("FAIL: non-aligned AXI MM FPGA addr 0x%llx\n",
+				(unsigned long long)pos);
+			return -EINVAL;
+		}
+
+		if (len_lsb != 0) {
+			dbg_tfr("FAIL: len %d is not a multiple of %d\n",
+				(int)count,
+				(int)engine->len_granularity);
```

Comment:
Linebreak right after opening brace is more preferable due to maintainability
```suggestion
			dbg_tfr(
				"FAIL:_ len %d is not a multiple of %d\n",
				(int)count,
				(int)engine->len_granularity);
```